File: src/dto/SubmitImagineDTO.ts
import { BaseSubmitDTO } from './BaseSubmitDTO';

/**
 * Imagine submission parameters
 */
export class SubmitImagineDTO extends BaseSubmitDTO {
  /** Prompt */
  prompt?: string;

  /** Base64 array of reference images */
  base64Array?: string[];

  /** Base64 (deprecated - use base64Array instead) */
  base64?: string;
}

File: src/main/java/com/github/novicezk/midjourney/dto/SubmitImagineDTO.java
package com.github.novicezk.midjourney.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;


@Data
@EqualsAndHashCode(callSuper = true)
public class SubmitImagineDTO extends BaseSubmitDTO {

	private String prompt;

	private List<String> base64Array;

	@Deprecated(since = "3.0", forRemoval = true)
	private String base64;

}
